File: sample-datacapture/DataCapture/RTRProgressView.m
// ABBYY® Real-Time Recognition SDK 1 © 2016 ABBYY Production LLC.
// ABBYY is either a registered trademark or a trademark of ABBYY Software Ltd.

#import "RTRProgressView.h"

@interface RTRProgressView ()

@property (nonatomic, weak) IBOutlet UIView* view;

@end

@implementation RTRProgressView

- (instancetype)initWithCoder:(NSCoder*)aDecoder
{
	self = [super initWithCoder:aDecoder];
	if(self != nil) {
		[self doInitRoutine];
	}
	return self;
}

- (instancetype)initWithFrame:(CGRect)frame
{
	self = [super initWithFrame:frame];
	if(self != nil) {
		[self doInitRoutine];
	}
	return self;
}

- (void)doInitRoutine
{
	NSString* className = NSStringFromClass([self class]);
	self.view = [[[self currentBundle] loadNibNamed:className owner:self options:nil] firstObject];

	NSAssert(self.view != nil, nil);
	[self addSubview:self.view];

	NSDictionary* views = @{ @"view" : self.view };
	NSArray* horizontalConstraints = [NSLayoutConstraint constraintsWithVisualFormat:@"H:|-0-[view]-0-|" options:0 metrics:nil views:views];

	NSArray* verticalConstraints = [NSLayoutConstraint constraintsWithVisualFormat:@"V:|-0-[view]-0-|" options:0 metrics:nil views:views];
	self.view.translatesAutoresizingMaskIntoConstraints = NO;

	[NSLayoutConstraint activateConstraints:horizontalConstraints];
	[NSLayoutConstraint activateConstraints:verticalConstraints];
}

- (NSBundle*)currentBundle
{
#if !TARGET_INTERFACE_BUILDER
	return [NSBundle mainBundle];
#else
	return [NSBundle bundleForClass:[self class]];
#endif
}

- (void)setProgress:(NSInteger)progress color:(UIColor*)color
{
	NSArray<UIView*>* rings = self.view.subviews;
	[rings enumerateObjectsUsingBlock:^(UIView* obj, NSUInteger idx, BOOL* stop) {
		obj.backgroundColor = (idx + 1 <= progress) ? color : [UIColor clearColor];
		obj.layer.borderColor = color.CGColor;
	}];
}

@end

File: sample-textcapture-swift/TextCapture/RTRProgressView.swift
// ABBYY® Real-Time Recognition SDK 1 © 2016 ABBYY Production LLC.
// ABBYY is either a registered trademark or a trademark of ABBYY Software Ltd.

import Foundation

@IBDesignable
class RTRProgressView: UIView {
	
	@IBOutlet weak var view: UIView?
	
	required init?(coder aDecoder: NSCoder)
	{
		super.init(coder: aDecoder)
		self.doInitRoutine()
	}
	
	required override init(frame: CGRect)
	{
		super.init(frame: frame)
		self.doInitRoutine()
	}
	
	public func setProgress(_ progress: Int, _ color: UIColor)
	{
		guard let rings = self.view?.subviews else {
			return
		}
		
		rings.enumerated().forEach { (index, view) in
			view.backgroundColor = (index + 1 <= progress) ? color : .clear
			view.layer.borderColor = color.cgColor
		}
	}
	
	public var currentBundle: Bundle {
		#if !TARGET_INTERFACE_BUILDER
			return Bundle.main
		#else
			return Bundle(for: type(of: self))
		#endif
	}
	
	public func doInitRoutine()
	{
		let className = NSStringFromClass(type(of: self))
		guard let views = self.currentBundle.loadNibNamed(className, owner: self, options: nil)
			, let view = views.first as? UIView else {
				print("View not loaded")
				return
		}
		
		self.addSubview(view)
		let viewsDic = ["view": view]
		
		let horizontalConstraints = NSLayoutConstraint.constraints(withVisualFormat: "H:|-0-[view]-0-|", metrics: nil, views: viewsDic)
		let verticalConstraints = NSLayoutConstraint.constraints(withVisualFormat: "V:|-0-[view]-0-|", metrics: nil, views: viewsDic)
		view.translatesAutoresizingMaskIntoConstraints = false
		NSLayoutConstraint.activate(horizontalConstraints + verticalConstraints)
		
		self.view = view
	}

}
